File: DevExpress VCL 23.2.6 Full Source/VCL/ExpressPDFViewer/Demos/Delphi/PageMerging/PageMergingMain.pas
unit PageMergingMain;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms, Dialogs, cxGraphics, cxControls,
  cxLookAndFeels, cxLookAndFeelPainters, dxSkinsCore, BaseForm, dxCore, dxCoreClasses, dxHashUtils, dxPDFCore,
  dxPDFBase, dxPDFText, dxPDFRecognizedObject, dxPDFDocument, dxCustomPreview, dxPDFDocumentViewer,
  dxPDFViewer, dxLayoutContainer, cxClasses, dxLayoutControl, dxLayoutControlAdapters, Menus, StdCtrls,
  cxButtons, dxLayoutLookAndFeels, dxLayoutcxEditAdapters, dxShellDialogs;

type
  TfrmPDFPageMerging = class(TfmBaseForm)
    dxPDFViewer1: TdxPDFViewer;
    dxLayoutItem1: TdxLayoutItem;
    btnOpen: TcxButton;
    btnAppend: TcxButton;
    dxLayoutItem3: TdxLayoutItem;
    dxLayoutItem4: TdxLayoutItem;
    btnSave: TcxButton;
    dxLayoutItem5: TdxLayoutItem;
    dxLayoutGroup1: TdxLayoutGroup;
    btnNew: TcxButton;
    dxLayoutItem2: TdxLayoutItem;
    procedure FormCreate(Sender: TObject);
    procedure btnNewClick(Sender: TObject);
    procedure btnOpenClick(Sender: TObject);
    procedure btnAppendClick(Sender: TObject);
    procedure btnSaveClick(Sender: TObject);
  protected
    procedure UpdateControls;
    procedure LoadDocument(const AFileName: string);
  end;

var
  frmPDFPageMerging: TfrmPDFPageMerging;

implementation

{$R *.dfm}

procedure TfrmPDFPageMerging.btnAppendClick(Sender: TObject);
var
  ADocument: TdxPDFDocument;
begin
  if OpenDialog.Execute then
  begin
    Screen.Cursor := crHourGlass;
    ADocument := TdxPDFDocument.Create;
    try
      ADocument.LoadFromFile(OpenDialog.FileName);
      dxPDFViewer1.Document.Append(ADocument);
    finally
      ADocument.Free;
      Screen.Cursor := crDefault;
    end;
  end;
  UpdateControls;
end;

procedure TfrmPDFPageMerging.btnNewClick(Sender: TObject);
begin
  dxPDFViewer1.Document.Clear;
  UpdateControls;
end;

procedure TfrmPDFPageMerging.btnOpenClick(Sender: TObject);
begin
  if OpenDialog.Execute then
    LoadDocument(OpenDialog.FileName);
  UpdateControls;
end;

procedure TfrmPDFPageMerging.btnSaveClick(Sender: TObject);
begin
  SaveDialog.DefaultExt := 'pdf';
  if SaveDialog.Execute then
    dxPDFViewer1.Document.SaveToFile(SaveDialog.FileName, True);
  UpdateControls;
end;

procedure TfrmPDFPageMerging.FormCreate(Sender: TObject);
begin
  LoadDocument('..\..\Data\PageMerging.pdf');
end;

procedure TfrmPDFPageMerging.UpdateControls;
begin
  btnSave.Enabled := dxPDFViewer1.IsDocumentLoaded;
end;

procedure TfrmPDFPageMerging.LoadDocument(const AFileName: string);
begin
  dxPDFViewer1.BeginUpdate;
  dxPDFViewer1.LoadFromFile(AFileName);
  dxPDFViewer1.OptionsZoom.ZoomFactor := 70;
  dxPDFViewer1.EndUpdate;
end;

end.

File: DevExpress VCL 23.2.6 Full Source/VCL/ExpressPDFViewer/Demos/CBuilder/PageMerging/PageMergingMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "PageMergingMain.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "BaseForm"
#pragma link "cxClasses"
#pragma link "cxLookAndFeels"
#pragma link "cxButtons"
#pragma link "cxControls"
#pragma link "cxGraphics"
#pragma link "cxLookAndFeelPainters"
#pragma link "dxCustomPreview"
#pragma link "dxLayoutContainer"
#pragma link "dxLayoutControl"
#pragma link "dxLayoutControlAdapters"
#pragma link "dxLayoutLookAndFeels"
#pragma link "dxPDFBase"
#pragma link "dxPDFCore"
#pragma link "dxPDFDocument"
#pragma link "dxPDFDocumentViewer"
#pragma link "dxPDFRecognizedObject"
#pragma link "dxPDFText"
#pragma link "dxPDFViewer"
#pragma link "dxShellDialogs"
#pragma resource "*.dfm"
TfrmPDFPageMerging *frmPDFPageMerging;
//---------------------------------------------------------------------------
void __fastcall TfrmPDFPageMerging::LoadDocument(const UnicodeString AFileName)
{
  dxPDFViewer1->BeginUpdate();
  dxPDFViewer1->LoadFromFile(AFileName);
  dxPDFViewer1->OptionsZoom->ZoomFactor = 70;
  dxPDFViewer1->EndUpdate();
}
//---------------------------------------------------------------------------
__fastcall TfrmPDFPageMerging::TfrmPDFPageMerging(TComponent* Owner)
	: TfmBaseForm(Owner)
{
  LoadDocument("..\\..\\Data\\PageMerging.pdf");
}
//---------------------------------------------------------------------------

void __fastcall TfrmPDFPageMerging::UpdateControls()
{
  btnSave->Enabled = dxPDFViewer1->IsDocumentLoaded();
};

void __fastcall TfrmPDFPageMerging::btnNewClick(TObject *Sender)
{
  dxPDFViewer1->Document->Clear();
  UpdateControls();
}
//---------------------------------------------------------------------------

void __fastcall TfrmPDFPageMerging::btnOpenClick(TObject *Sender)
{
  if (OpenDialog->Execute())
	LoadDocument(OpenDialog->FileName);
  UpdateControls();
}
//---------------------------------------------------------------------------

void __fastcall TfrmPDFPageMerging::btnSaveClick(TObject *Sender)
{
  if (SaveDialog->Execute())
	dxPDFViewer1->Document->SaveToFile(SaveDialog->FileName, True);
  UpdateControls();
}
//---------------------------------------------------------------------------

void __fastcall TfrmPDFPageMerging::btnAppendClick(TObject *Sender)
{
  TdxPDFDocument *ADocument;

  if (OpenDialog->Execute())
  {
	Screen->Cursor = crHourGlass;
	ADocument =  new TdxPDFDocument();
	try
	{
	  ADocument->LoadFromFile(OpenDialog->FileName);
	  dxPDFViewer1->Document->Append(ADocument);
	}
	__finally
	{
	  ADocument->Free();
	  Screen->Cursor = crDefault;
	}
  }
  UpdateControls();
}
//---------------------------------------------------------------------------



